### DISCUSSION TITLE
How to replace gnome-flashback screensaver?

### DISCUSSION THREAD

**Discussion by gdeflaux** [regolith-linux/regolith-desktop#987] [Q&A]
Hey Regolith Community!

I'd like to know if it's possible to replace `gnome-flashback screensaver` with another program. I've managed to disable it with `gsettings` as mentioned [here](https://regolith-desktop.com/docs/howtos/change-lockscreen/#1-configure-gnome-flashback-to-disable-built-in-lockscreen), but then, the screen never turns off, hence never locks either when the computer is unused.

What I would like to achieve is:

- Disable gnome-flashback screensaver
- Get the screen to turn off after X minutes as specified in the settings
![image](https://github.com/regolith-linux/regolith-desktop/assets/223608/c8257964-ba5e-47b5-a6a8-4d897dee36f6)
- Use the same locking script specified in my `Xresources` key `wm.program.lock` to lock the screen when it turns off.

Thanks for your help!

PS: Coming back to Regolith after a while and it rocks!

**Comment by kgilmer** [comment]
Hi @gdeflaux , nice to see a familiar face :).  Hmm, if the `gsettings` change isn't disabling [gnome-flashback's screensaver](https://gitlab.gnome.org/GNOME/gnome-flashback/-/issues/18) I think that would be an upstream bug.  I believe this used to work..

**Comment by gdeflaux** [comment]
Hey!

The `gsettings` change is working. But then there is no more screensaver, and the screen never turns off. So I am curious: how can you change the screensaver program?

**Comment by NicolasDenier** [comment]
Hello @gdeflaux , I encountered the same need.  
My fix is to use `xautolock`, a program to lock after x min of inactivity with a custom locker script.  
In my i3 config: `xautolock -time 5 -secure -locker $wm.program.lock`  
Hope this is what you want.

**Comment by gdeflaux** [comment]
Hey @NicolasDenier. Thanks for tip. That works. Here's my actual script: 

```
#!/bin/bash

SS_FILE=/tmp/screen.png

/usr/bin/scrot $SS_FILE
/usr/bin/convert $SS_FILE -scale 10% -scale 1000% $SS_FILE
i3lock -i $SS_FILE
rm $SS_FILE

# pgrep makes sure that the screens turns off
# only if i3lock is still running, i.e. the 
# computer was not unlcoked before sleep times
# out
sleep 20; pgrep i3lock && xset dpms force off
```

[Credit](https://faq.i3wm.org/question/83/how-to-run-i3lock-after-computer-inactivity.1.html)

There are still a few issues to work out such as locking the screen on suspend (laptop).
